documentation/guide-spring-data.asciidoc

```diff
@@ -0,0 +1,56 @@
+There are many discussions in the devonfw community about Spring Data vs classic DAO approach. Ultimately, we support both.
+
+== Spring Data in Quarkus
+
+Quarkus currently supports many features of Spring Data JPA. A detailed guide can be found https://quarkus.io/guides/spring-data-jpa[here].
+
+=== **Repository**
+To implement a repository, the interface has to extend one of the following Spring Data repositories:
+
+    * org.springframework.data.repository.Repository
+    * org.springframework.data.repository.CrudRepository
+    * org.springframework.data.repository.PagingAndSortingRepository
+    * org.springframework.data.jpa.repository.JpaRepository
+
+These Spring Data Repositories provide some basic implementations for accessing data. For example, the `CrudRepository` interface provides methods for returning all instances of a type `findAll` or returning an instance by its ID (`findById`).
+Methods of repository interfaces that follow Spring Data conventions are implemented automatically. This means that you do not have to write code for them. Examples:
+
+    * `findByName` (if there is a name attribute)
+    * `findByNameOrderByAge` (if there is a name and an age attribute)
+
+The generated repositories are also registered as beans so they can be injected into any other bean. Furthermore the methods that update the database are automatically annotated with @Transactional.
+
+You can also fine-tune the repository definition for your own need by defining repository interfaces to cherry-pick methods from any of the supported Spring Data repository interfaces without having to extend those interfaces. It's helpful when for example a repository needs to use some methods from `CrudRepository` but it's undesirable to expose the full list methods of said interface.
+
+In addition, repositories can be enriched with additional functionality or override the default implementations by using repository fragments.
+
+=== **Query**
+
+Quarkus supports the definition of static PSQL queries with the `@Query` annotation. Examples:
+```
+@Query("select a from Animal a where name = :name")
+	Animal findByName(@Param("name") String name);
+```
+Quarkus supports the use of the Criteria API in both native and JVM mode to define dynamic queries.
+
+QuerySQL, contrary to what is claimed in the https://quarkus.io/guides/spring-data-jpa[Quarkus Spring Data guide], works in both native and JVM mode to define dynamic queries.
+
+=== **Pagination**
+Quarkus Spring Data offers paging support. It seems to be fully functional only in JVM mode.
+
+== Limitation
+
+* Native and named queries are not supported using `@Query` annotation. You will receive something like `Build step io.quarkus.spring.data.deployment.SpringDataJPAProcessor#build threw an exception: java.lang.IllegalArgumentException: Attribute nativeQuery of @Query is currently not supported.`
+* Paging support does not work in native mode.
```

Comment:
This is IMHO a critical limitation.
Can you do some research and find a github issue or something about this so we can track the status or even see a roadmap indicating when to expect a fix for this?

Reply:
If paging is part of the offiical documentation of quarkus spring-data-jpa support I am slightly confused:
https://quarkus.io/guides/spring-data-jpa#derived-query-methods

I can not believe that the wrote this documentation and never ever tested this...

Reply:
https://quarkus.io/guides/spring-data-jpa#what-is-currently-unsupported

Reply:
These are the limitations @GuentherJulian determined after testing/evaluating the feature [here](https://github.com/devonfw-forge/devonfw-microservices/issues/37). As that the pagination support seems to work only in JVM mode, NOT in native mode

Reply:
I will test it again...

Reply:
Check the paging support again. See https://github.com/devonfw-forge/devonfw-microservices/issues/37#issuecomment-888980547